src/lava/magma/core/process/ports/ports.py

```diff
@@ -46,11 +46,11 @@ def __init__(self, shape: ty.Tuple):
         self.out_connections: ty.List[AbstractPort] = []
```

Comment:
I very much like the docstring of AbstractPort! However, I would add a few information:
Maybe a first sentence like "Ports are the main building block of Lava to exchange data between different processes."

For the sentence "Sub classes of AbstractPort only facilitate connecting to other compatible ports"
I would state the actual connections that are possible (OutPort -> Inport; VarPort -> RefPort).

"a port may have zero or more input and output connections"
This sounds as if every Port can have both input and output connections. But an InPort cannot have one or multiple Input connections.

"These connections, imply an a-cyclic graph structure that allows the compiler to infer connections between processes."
This sounds as if we cannot build cyclig graphs (like RNN) in Lava. Not sure what is exactly meant with this statement?

Reply:
This is the most abstract description and definition of a Port we have in Lava. In/Out or Ref/Var Ports are much more specialized and should not be mentioned here at all.
Maybe someone adds different Port types in the future.

Also, this docstring is from Andreas I think, I have not touched docstrings in this file and have little knowledge of what is going on there. So I prefer to not alter the docstring in this file in this PR.